gcs-app/gcs.py

```diff
@@ -19,14 +19,14 @@ def upload_blob(self, bucket_name, file_name, contents):
         blob = bucket.blob(file_name)
         blob.upload_from_string(contents)
         print(
-            "File {} uploaded to {}.".format(
-                file_name, file_name
+            "File {} uploaded to bucket {} as file {}.".format(
+                file_name, bucket_name, file_name
             )
         )
 
 if __name__ == '__main__':
     gcs = Gcs()
     bucket = os.getenv('BUCKET')
-    job_name = os.getenv('JOB_NAME')
+    job_name = os.getenv('JOB_NAME') + f"-{str(uuid.uuid4())}"
```

Comment:
You shouldn't concatenate f-strings.  Also, a JOB_NAME and a full uuid4() may be longer than the allowed k8s job name.   Cut off at 8 chars.

 ```python
   job_name = f"{os.getenv('JOB_NAME')}-{str(uuid.uuid4())[:8]}"
```

Reply:
This is not a job name. It's the name of the file that will be created in the bucket

Reply:
> This is not a job name. It's the name of the file that will be created in the bucket

Ok, so we should also rename this to file_name so that it makes sense. And you still shouldn't concat f-strings because it's ugly.

Reply:
Can we use this instead? Concat the name with a timestamp. This is to ensure we don't overwrite the already existing files in the bucket.

Reply:
My problem is that there shouldn't be a create_job_name() function in the K8s class. Why does generating a job name have to do with a class created for interacting with k8s?  What happens if the user doesn't want to use a job name with a time stamp or a different format?

It doesn't belong there and makes for code that is not properly abstracted. The issue is that you created a class, so therefore the class needs to be better designed.

Reply:
We also are creating 1 job with multiple completions, so all env vars are going to be the same. It is up to gcs.py to generate the filenames. 

Additionally, when you want unique values, you can't rely on time stamps. That is the purpose of uuid.

Reply:
This code snippet is in GCS. we are taking the `job_name` and adding a dynamic suffix to it so that the file create in the bucket is different for every completion.
The ideas I have for suffixes are a timestamp or a uuid. we can use like 5 characters of the uuid instead of the whole string like is set now.

Reply:
I understand what you are trying to do. I was trying to explain to you why using the function in the k8s should not be done.  I've also tried to give some background on my thought process. I've already provided the solution in my original comment.  At this point just going back and forth is starting to feel like a waste of time.